app/src/main/java/org/p2p/wallet/history/ui/token/TokenHistoryFragment.kt

```diff
@@ -165,6 +165,13 @@ class TokenHistoryFragment :
 
     override fun showPagingState(newState: PagingState) {
         historyAdapter.setPagingState(newState)
+        with(binding) {
+            shimmerView.isVisible = newState == PagingState.InitialLoading
+            refreshLayout.isVisible = newState != PagingState.InitialLoading
+            emptyStateLayout.isVisible = newState == PagingState.Idle && historyAdapter.isEmpty()
+            historyRecyclerView.isVisible =
+                (newState == PagingState.Idle && !historyAdapter.isEmpty()) || newState == PagingState.Loading
```

Comment:
Why do you make such complex conditions for simple states? 
You'd rather to check each state and update UI according to it. 
This one is hard to support.

The best way will be controlling the UI from presenter

Reply:
can we refactor this later ?